utils.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2019 The ClangBuiltLinux Authors
+# Description: Common helper functions
+
+import hashlib
+import pathlib
+import shutil
+import subprocess
+
+
+# Create a gitignore that ignores all files in a folder
+def create_gitignore(folder):
```

Comment:
why is this needed?

Reply:
A user may choose to install the toolchain within the repository but not in `usr` so we can't add it to the root `.gitignore`.

Reply:
Sounds like that justification should go in the function's comment.